src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -76,15 +77,185 @@ public void registered(SchedulerDriver driver, FrameworkID frameworkId, MasterIn
+        liveState.getStagingTasksSize()));
+
+    if (!isStagingState(status)) {
+      liveState.removeStagingTask(status.getTaskId());
+    }
+
+    if (isTerminalState(status)) {
+      liveState.removeRunningTask(status.getTaskId());
+      persistentState.removeTaskId(status.getTaskId().getValue());
+      if (liveState.reconciliationComplete()) {
+        correctCurrentPhase();
+      }
+
+    } else if (isRunningState(status)) {
+      liveState.updateTaskForStatus(status);
+
+      log.info(String.format("Current Acquisition Phase: %s", liveState
+          .getCurrentAcquisitionPhase().toString()));
+
+      switch (liveState.getCurrentAcquisitionPhase()) {
+        case RECONCILING_TASKS :
+          if (liveState.reconciliationComplete()) {
+            reconcilePersistentState();
+            correctCurrentPhase();
+          }
+          break;
+        case JOURNAL_NODES :
+          if (liveState.getJournalNodeSize() == conf.getJournalNodeCount()) {
+            correctCurrentPhase();
+          }
+          break;
+        case START_NAME_NODES :
+          if (liveState.getNameNodeSize() == (HDFSConstants.TOTAL_NAME_NODES)) {
+            // TODO move the reload to correctCurrentPhase and make it idempotent
+            reloadConfigsOnAllRunningTasks(driver);
+            correctCurrentPhase();
+          }
+          break;
+        case FORMAT_NAME_NODES :
+          if (!liveState.isNameNode1Initialized()
+              && !status.getMessage().equals(HDFSConstants.NAME_NODE_INIT_MESSAGE)) {
+            sendMessageTo(
+                driver,
+                liveState.getFirstNameNodeTaskId(),
+                liveState.getFirstNameNodeSlaveId(),
+                HDFSConstants.NAME_NODE_INIT_MESSAGE);
+          } else if (liveState.isNameNode1Initialized()
+              && !liveState.isNameNode2Initialized()) {
+            sendMessageTo(
+                driver,
+                liveState.getSecondNameNodeTaskId(),
+                liveState.getSecondNameNodeSlaveId(),
+                HDFSConstants.NAME_NODE_BOOTSTRAP_MESSAGE);
+          } else if (liveState.isNameNode1Initialized()
+              && liveState.isNameNode2Initialized()) {
+            correctCurrentPhase();
+          }
+          break;
+        case DATA_NODES :
+          break;
+      }
+    } else {
+      log.warn(String.format("Don't know how to handle state=%s for taskId=%s",
+          status.getState(), status.getTaskId().getValue()));
+    }
+  }
+
+  @Override
+  public void resourceOffers(SchedulerDriver driver, List<Offer> offers) {
+    log.info(String.format("Received %d offers", offers.size()));
+
+    if (liveState.getStagingTasksSize() != 0) {
+      log.info("Declining offers because tasks are currently staging");
+      for (Offer offer : offers) {
+        driver.declineOffer(offer.getId());
+      }
+    } else {
+      boolean acceptedOffer = false;
+      for (Offer offer : offers) {
+        if (acceptedOffer) {
```

Comment:
Currently accepting only one offer per resourceOffers() call. Great for keeping the logic simple now, but in the future, we could accept multiple offers if necessary/desired, e.g. launch all JNs or both NNs, or lots of DNs in the same offer cycle, rather than waiting 5 seconds before getting reoffered the additional resources.